### ISSUE TITLE
absent_score

### ISSUE THREAD

**Issue by mnslarcher** [Lightning-AI/metrics#1445]
### Motivation

In previous versions of torchmetrics I was using absent_score=1.0, now I don't see this parameter anymore.

Considering a case like:
```
preds = torch.ones((10,))
y = torch.ones((10,))
metric = torchmetrics.JaccardIndex(task="multiclass", num_classes=3)
metric_noavg = torchmetrics.JaccardIndex(task="multiclass", num_classes=3, average="none")
print(metric(preds, y))
print(metric_noavg(preds, y))
tensor(0.3333)
tensor([0., 1., 0.])
```

It seems wrong to me not to have 100% scoring when the model correctly predicts 100% of the instances. I have many cases where in a segmentation task I don't have all the classes and I think the scoring on these images is confusing.

### Pitch

I would like or to have back the absent_score parameter or to have a behavior where if the model is 100% right the score is 1.

**Comment by mnslarcher** [comment]
> preds = torch.ones((10,))
> y = torch.ones((10,))
> metric = torchmetrics.JaccardIndex(task="multiclass", num_classes=3)
> metric_noavg = torchmetrics.JaccardIndex(task="multiclass", num_classes=3, average="none")
> print(metric(preds, y))
> print(metric_noavg(preds, y))
> tensor(0.3333)
> tensor([0., 1., 0.])

Thanks for the answer @SkafteNicki !

I think micro is a more natural extension of the binary case for the reason already mentioned and for consistency:

```
preds = torch.ones((10,))
y = torch.ones((10,))
metric_binary = torchmetrics.JaccardIndex(task="binary")
metric_multiclass_macro = torchmetrics.JaccardIndex(task="multiclass", num_classes=2)
metric_multiclass_micro = torchmetrics.JaccardIndex(task="multiclass", num_classes=2, average="micro")
print(metric_binary(preds, y))
print(metric_multiclass_macro(preds, y))
print(metric_multiclass_micro(preds, y))
tensor(1.)
tensor(0.5000)
tensor(1.)
```
binary and multiclass when num_classes=2 should be equivalent (at least in the default setting), what do you think?

**Comment by SkafteNicki** [comment]
@mnslarcher until v0.10 the default for `average` was actually `"micro"` however, we often encountered that users was complaining about the scores being the same for `accuracy`, `precision`, `recall` e.g.:
```python
from torchmetrics.classification import MulticlassAccuracy, MulticlassRecall, MulticlassPrecision
from torchmetrics import MetricCollection
import torch
metrics = MetricCollection(
  MulticlassAccuracy(num_classes=2, average='micro'), 
  MulticlassRecall(num_classes=2, average='micro'), 
  MulticlassPrecision(num_classes=2, average='micro')
)
metrics(torch.randint(2, (20,)), torch.randint(2, (20,)))
# prints
# {'MulticlassAccuracy': tensor(0.3500),
#  'MulticlassRecall': tensor(0.3500),
#  'MulticlassPrecision': tensor(0.3500)}
```
which is correct as the scores collapse to the same definition under the `average="micro"` reduction. For that reason we changed the default to `"macro"` in v0.10, which does not have this "problem", but still have other problems (as you are describing). Thus, again there is really no right way to do it. `sklearn` goes around the problem by forcing you to set the `average` argument yourself in the multiclass case, which I do not think is the solution for `torchmetrics`.

I am thinking to keep the default `"macro"` for the moment. If more users report that it is confusing, then we can reconsider what the default should be.

**Comment by mnslarcher** [comment]
Thanks for the answer @SkafteNicki ! Let's see what other users think then, maybe in the meantime it would be helpful to make this more explicit in the doc if not forcing the user to choose like in sklearn

**Comment by mnslarcher** [comment]
@SkafteNicki : 

I notice that `micro` is not equivalent of `macro` + `absent_score=1.0` (so is not an alternative for `absent_score` that is still useful and I would set to `1.0` by default). 

In particular `micro` is not good for an imbalanced problem where `macro` + `absent_score=1.0` is OK and it has the good properties already written above.

For example (I use accuracy just because is easier to interpret):

```
from torchmetrics import Accuracy
import torch

pred = torch.zeros(1000)
target = torch.zeros(1000)
target[:10] = 1

acc_none = Accuracy(task="multiclass", num_classes=3, average="none")
acc_macro = Accuracy(task="multiclass", num_classes=3, average="macro")
acc_micro = Accuracy(task="multiclass", num_classes=3, average="micro")
# acc_macro_absent_score = JaccardIndex(task="multiclass", num_classes=3, average="macro", absent_score=1.0)

print(acc_none(pred, target))
print(acc_macro(pred, target))
print(acc_micro(pred, target))
# print(acc_macro_absent_score(pred, target))

# none
tensor([1., 0., 0.])

# macro
# The model is penalized for the absence of a class in the target
tensor(0.3333)

# micro
# too much weight to the highest frequency class
tensor(0.9900) 

# macro + absent_score=1.0
# assigns the model a score of 100% for the absent class that did not predict (correctly)
# don't "overweight" the highest frequency class
# tensor(0.6333).
```

So the request is, can we have the `absent_score` back (possibly set to `1.0` by default)?
